### PR TITLE
Lens style

### PR DESCRIPTION
**Pull request by cchalmers**
See diagrams/diagrams-core#80. Includes a bunch of lenses onto attributes of a `Style`.

The only real issue is lenses that include a `Recommend` are always "committed" (this is how the previous `styleFillTexture` setter was). This is almost always what you want, otherwise styles set with a lens could be accidentally overwritten. These lenses have warnings and a valid `R` variant that includes the `Recommend`.

The 3D attributes lenses haven't been added these lenses because the povray backend needs changes, I'll do that later if this gets merged.

Other then that this should be ready to merge.

### CODE REVIEW

**Comment by cchalmers** [comment]
I've added `Iso`s in the form `_Attribute :: Iso AttributeA Attribute` and an `_attribute :: Lens' (Style v n) Attribute` for most attributes. Some of them are `Maybe Attribute` and some of them have a `_attributeR :: Lens' (Style v n) (Recommend Attribute)` variant. I've also added some lenses / prisms for `Recommend`.

The changes shouldn't break any backends (except for povray which I'll fix). Overall I'm reasonably happy with it, I think it's ready to merge. What do people think?

File: src/Diagrams/Attributes.hs [hunk 1]
```diff
@@ -338,3 +410,39 @@ lineMiterLimit = applyAttr . LineMiterLimit . Last
...
+_Commit :: Prism' (Recommend a) a
+_Commit = prism' Commit $ \case (Commit a) -> Just a; _ -> Nothing
+
+-- | Lens onto the value inside either a 'Recommend' or 'Commit'. Unlike
+--   'committed', this is a valid lens.
+_recommend :: Lens (Recommend a) (Recommend b) a b
+_recommend f (Recommend a) = Recommend <$> f a
+_recommend f (Commit a)    = Commit <$> f a
+
+-- | Lens onto weather something is committed or not.
```
Comment (bergey) [inline src/Diagrams/Attributes.hs]: `whether`, not weather

Comment (fryguybob) [inline src/Diagrams/Attributes.hs]: If only we had a lens into weather...

**Comment by bergey** [comment]
Looks good to me.

**Comment by byorgey** [comment]
Looks good to me too.

**Pull request merged**